src/main/java/io/confluent/connect/jdbc/dialect/OracleDatabaseDialect.java

```diff
@@ -77,14 +89,80 @@ protected String checkConnectionQuery() {
   }
 
   @Override
+  public StatementBinder statementBinder(
```

Comment:
Hmm, this is unfortunate that we need this other method to enable the new functionality. I think this means that the rollout needs an additional phase:
1. this PR to add the deprecated functions
2. migration of all of the leaf dialects including adding this override
3. Adding the override to Generic, removing the old methods
4. Removing all of these unnecessary overrides from the leaf dialects

It seems unavoidable, since we can't provide a non-null tableDefinition by default without erasing a downstream dialect's functionality.